functions/routes/categoryRoutes.js

```diff
@@ -0,0 +1,130 @@
+const router = require('express')
+                    .Router();
+const categoriesRef = require('firebase-admin')
+                            .firestore()
+                            .collection('category');
+const Category = require('../model/Category')
+
+                            
+router
+    .route('/')
+    .get(getAllCategories)
+    .post(addNewCategory)
+    .put(updateCategory)
+    .delete(deleteCategory);
+
+
+function getAllCategories(req, res) {
+    categoriesRef
+        .get()
+        .then( (categories) => {
+            return res.status(200).json({
+                success: true,
+                message: 'List of All Categories Received Successfully',
+                data: {
+                    categories: categories
+                                    .docs
+                                    .map(doc => Category
+                                        .getCategoryConverter()
+                                        .fromFirestore(doc))
```

Comment:
Later we can put a try catch block here too...
To check whether categoryDoc get converted into Category Class or not.

Not important Right Now